.circleci/config.yml

```diff
@@ -32,7 +32,7 @@ jobs:
       - run:
           name: golangci-lint
           command: |
-            curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.23.6
+            curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.41.1
```

Comment:
Maybe we should move to a different way of downloading this. I'm not a huge fan of the unverified curl bash

Reply:
It is unfortunately the recommended way to do it: https://golangci-lint.run/usage/install/#other-ci

Reply:
Can we pull it from releases and verify the sha?
https://github.com/golangci/golangci-lint/releases

Reply:
Ok my recent change does that instead. I made the version and checksum env vars so it's not too cumbersome to update.